src/tests/Microsoft.PowerFx.Interpreter.Tests/CollectFunctionTests.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.PowerFx.Core.App.ErrorContainers;
+using Microsoft.PowerFx.Core.Binding;
+using Microsoft.PowerFx.Core.Errors;
+using Microsoft.PowerFx.Core.Functions;
+using Microsoft.PowerFx.Core.Functions.FunctionArgValidators;
+using Microsoft.PowerFx.Core.Localization;
+using Microsoft.PowerFx.Core.Tests;
+using Microsoft.PowerFx.Core.Types;
+using Microsoft.PowerFx.Core.Utils;
+using Microsoft.PowerFx.Interpreter;
+using Microsoft.PowerFx.Syntax;
+using Microsoft.PowerFx.Types;
+using Xunit;
+using static Microsoft.PowerFx.Core.Localization.TexlStrings;
+
+namespace Microsoft.PowerFx.Interpreter.Tests
+{
+    public class CollectFunctionTests : PowerFxTest
```

Comment:
>CollectFunctionTests [](http://example.com/codeflow?start=17&length=20)

See Set tests for some inspiration. 

More test ideas:
- verify that Collect() fails in a non-behavior context. 

- Type checking - what if we pass the wrong type to collect (too few fields, too many fields, not a record, etc). 

- PowerApps lets pass multiple records - this is determined by CheckInvocation, which we copied from PA.  We should either 
a) reject that as an error, 
b) support it. 
Collect(_c, {r:1}, {r:2})

- will need to confirm corner case behavior, ie, what if we pass errors, etc.  (We can probably do that in a followup PR, just to unblock usage now) 



- trying to collect on an Immutable object. We should get an error

Reply:
Also, type checking with marshalled types on both sides.

Reply:
fyi

Reply:
After some changes:

> verify that Collect() fails in a non-behavior context.

Throw an exception


> Type checking - what if we pass the wrong type to collect (too few fields, too many fields, not a record, etc).

All these cases will return an exception

> - PowerApps lets pass multiple records - this is determined by CheckInvocation, which we copied from PA. We should either
> 
> a) reject that as an error,
> b) support it.
> Collect(_c, {r:1}, {r:2})

For now, limiting argument numbers to two: table and record


> will need to confirm corner case behavior, ie, what if we pass errors, etc. (We can probably do that in a followup PR, just to unblock usage now)

I'll wait further interactions on this.

> trying to collect on an Immutable object. We should get an error

This test case has been added.